lib/gsockaddr.c

```diff
@@ -529,6 +529,21 @@ static GSockAddrFuncs unix_sockaddr_funcs =
   .format = g_sockaddr_unix_format
 };
 
+/*
+  SUN_LEN is not a POSIX standard, thus not available on all platforms.
+  If it is available we should rely on it. Otherwise we use the formula
+  from the Linux man page.
+*/
+static int
+_calculate_salen(GSockAddrUnix *addr)
+{
+#ifdef SUN_LEN
+  return SUN_LEN(&(addr->saun));
+#else
+  return sizeof(addr->saun) - sizeof(addr->saun.sun_path) + strlen(addr->saun.sun_path) + 1;
+#endif
+}
+
```

Comment:
So we have a platform where `SUN_LEN` is not defined? :)
I'd define `SUN_LEN` in a `compat` header.
What do you think?

an alternative impl.:
http://lists.llvm.org/pipermail/lldb-commits/Week-of-Mon-20151026/024736.html
Does this have the same result?